app/controllers/suggestions_controller.rb

```diff
@@ -0,0 +1,54 @@
+# require '/lib/tunes_takeout_wrapper'
+class SuggestionsController < ApplicationController
+  skip_before_action :require_login, only: :index
+
+  def index
+    @suggestions = extract_suggestions(TunesTakeoutWrapper.top(4))
+    #shows top 20 suggestions, ranked by total number of favorites
+    if current_user
+      @spotify_user = RSpotify::User.find(@current_user.uid)
+      @user = current_user
+      @image = @spotify_user.images[0]["url"]
+    end
+    # if current_user
+    #   @favorites = TunesTakeoutWrapper.user_favorites(@user.user_id)
+    # end
+  end
```

Comment:
I'm noticing that you have "if current_user" logic both here and in the partial you render. I'd recommend consolidating this logic in the controller. You could set a variable (like ```user_logged_in = false```) in the controller, then render different partials depending on whether or not the user is logged in.